Source/WarpX.cpp

```diff
@@ -1002,15 +1002,26 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
 #   endif
     // Check whether the option periodic, single box is valid here
     if (fft_periodic_single_box) {
-        AMREX_ALWAYS_ASSERT_WITH_MESSAGE( geom[0].isAllPeriodic() && ba.size()==1 && lev==0,
-        "The option `psatd.periodic_single_box_fft` can only be used for a periodic domain, decomposed in a single box.");
+#   ifdef WARPX_DIM_RZ
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+            geom[0].isPeriodic(1)          // domain is periodic in z
+            && ba.size() == 1 && lev == 0, // domain is decomposed in a single box
+            "The option `psatd.periodic_single_box_fft` can only be used for a periodic domain, decomposed in a single box");
+#   else
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+            geom[0].isAllPeriodic()        // domain is periodic in all directions
+            && ba.size() == 1 && lev == 0, // domain is decomposed in a single box
+            "The option `psatd.periodic_single_box_fft` can only be used for a periodic domain, decomposed in a single box");
+#   endif
```

Comment:
@dpgrote @RemiLehe Here I had to place the directives `#ifdef WARPX_DIM_RZ` etc. outside of the macro `AMREX_ALWAYS_ASSERT_WITH_MESSAGE`, because otherwise the compiler would issue a warning stating that placing directives inside a macro argument is not portable.